src/Integration.Vsix.UnitTests/InfoBar/InfoBarManagerTests.cs

```diff
@@ -303,6 +309,60 @@ public void AttachInfoBarWithButtons_ButtonClickedEvent_RaisedWithTheRightButton
             eventHandler.VerifyNoOtherCalls();
         }
 
+        #region MainWindow tests
+        // The ToolWindow tests above cover the common implementation of gold bar functionality.
+        // These MainWindow tests just cover a few extra main window specific cases.
+
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarToMainWindowAndDetach_ArgChecks()
+        {
+            var testSubject = CreateTestSubject();
+
+            Action act = () => testSubject.AttachInfoBarToMainWindow(null, SonarLintImageMoniker.OfficialSonarLintMoniker);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("message");
+
+            act = () => testSubject.AttachInfoBarToMainWindow("a message", SonarLintImageMoniker.OfficialSonarLintMoniker, null);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("buttonTexts");
+        }
+
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarToMainWindowAndDetach_AttachAndDetach()
+        {
+            var sp = new ConfigurableServiceProvider();
+            sp.RegisterService(typeof(SVsInfoBarUIFactory), new ConfigurableVsInfoBarUIFactory());
+
+            var host = new ConfigurableVsInfoBarHost();
+            RegisterMainWindowHostWithShell(sp, host);
+
+            var testSubject = new InfoBarManager(sp);
```

Comment:
The test class does have a `CreateTestSubject` method, but it does extra work. I didn't want to refactor/clean-up the existing tests as part of this PR as that would get confusing.

(I actually started doing the cleanup separately this morning until I noticed the problem with the VS2019 CI build)

Reply:
Yea agree, I thought about it too but I think its better to have a "clean up" ticket somewhere and put it in the backlog.

Reply:
Yep, that's the alternative. I was experimenting with a more "clean as you code" approach. I didn't want to pollute the existing PR with unrelated cleanup, but the trouble with creating a ticket is that we seldom get around to them. Also, since I've just been working on the tests I'm more familiar with them.

Rather than creating a ticket, I was going to spend a maximum of fifteen/twenty minutes doing some low-risk cleanup, and then send it for review. If that approach worked, we could try allocating a time budget for each sprint for miscellaneous cleanup e.g. two hours, to make sure we don't lose focus on the main purpose of the sprint. Something to discuss offline.